apis/v1alpha2/httproute_types.go

```diff
@@ -705,17 +705,25 @@ type HTTPPathModifier struct {
 	// Type defines the type of path modifier.
 	//
 	// <gateway:experimental>
-	// +kubebuilder:validation:Enum=Absolute;ReplacePrefixMatch
+	// +kubebuilder:validation:Enum=ReplaceFullPath;ReplacePrefixMatch
```

Comment:
Since you really don't have any other fields than union members in that struct, I don't think the discriminator is useful at all.

Reply:
^ this was me, I posted with a temporary account ;-)

Reply:
I think this adds room for growth. The goal here is to add additional types in the future. Since we already have 14 implementations and integrations, and have more on the way, it's very likely that at least some of these will trail the version of the API deployed in the cluster. A discriminator field allows those implementations to communicate that they don't support a specific type, instead of just seeing an empty struct and not knowing what to do with it.

Reply:
Yeah, they don't get a lot more value from seeing the struct empty vs seeing the discrimator set to a value that doesn't make sense (but they get a little value). In both cases, they can manually clear everything/set the discriminator to the right value.

Reply:
I chatted with @thockin about this, and while I think he still has significant reservations about using a union discriminator, I think he's on board with starting with one. The idea is that it's impossible to retroactively add a discriminator, but we could always make a discriminator optional and/or provide a smart default in the future. (Please correct me if I misunderstood anything).